website/utils/test/url_utils.test.ts

```diff
@@ -33,19 +35,46 @@ const defaultSettings: ViewerChannelSetting = {
 
 //// VALUE PARSING ///////////////////////////////////////
 
+describe("LEGACY_CONTROL_POINTS_REGEX", () => {
+  it("accepts single control points", () => {
+    const data = "1:0.5:ffffff";
+    expect(LEGACY_CONTROL_POINTS_REGEX.test(data)).toBe(true);
+  });
+
+  it("accepts multiple control points", () => {
+    const data = "1:0.5:ff0000,128:0.7:ffff00,255:1:ff0000";
+    expect(LEGACY_CONTROL_POINTS_REGEX.test(data)).toBe(true);
+  });
+
+  it("accepts negative numbers", () => {
+    const data = "-1:0.5:ff0000,-255:1:ff0000";
+    expect(LEGACY_CONTROL_POINTS_REGEX.test(data)).toBe(true);
+  });
+
+  it("allows 'w' as a placeholder for color strings", () => {
```

Comment:
also suggest allowing `0` for black, and `1` or (maybe more appropriately) `-1` for white, as well.  Something about that feels more consistent than having a non-numeric letter character.  See what you think. If not today, then we can consider it for a future PR.

Reply:
I'll replace "w" with 1, but I think anymore than that is out of scope for now. I added a related TODO about refactoring the regular expressions to be more composable.

Reply:
There's a few commented-out code blocks in here that hint at that change!